src/notifications/index.ts

```diff
@@ -145,22 +147,47 @@ export async function storeNotificationBundleV2(
   const notification = generatedMaps[0] as NotificationV2;
   const uniqueKey = generateUserNotificationUniqueKey(notification);
 
+  const userPreferences = new Map<string, User>();
+  const userChunks: string[][] = [];
+  const chunkSize = 500;
+
+  for (let i = 0; i < bundle.userIds.length; i += chunkSize) {
+    userChunks.push(bundle.userIds.slice(i, i + chunkSize));
+  }
+
+  for (const chunk of userChunks) {
+    const users = await entityManager.getRepository(User).find({
+      select: ['id', 'notificationFlags'],
+      where: { id: In(chunk) },
+    });
```

Comment:
Wander if we can do at insert time, eg. with insert into statement where we query users with preference?